shopping-list-using-service/src/app/add-item/add-item.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { AddNewItemService } from '../add-new-item.service';
+import { FormsModule } from '@angular/forms';
+
+@Component({
+  selector: 'app-add-item',
+  standalone: true,
+  imports: [FormsModule],
+  templateUrl: './add-item.component.html',
+  styleUrl: './add-item.component.scss'
+})
+export class AddItemComponent {
+  newItem: string='';
+  countvalue!:number;
+  
+  constructor(private AddItemService:AddNewItemService) {}
+
+  ngOnInit()
+  {
+    this.AddItemService.SetCountService$.subscribe(count=>this.setCount(count));
```

Comment:
better to unsubscribe to avoid memory leak. use takeUntilDestroyed with destroyRef